example/src/App.js

```diff
@@ -10,22 +11,46 @@ export default function App({ initialLocation }) {
 }
 
 function Page() {
-  const { matchRoutes } = useRouting()
-  const routes = routeMap
+  const location = useLocation()
+  const pageRoutePath = usePageRoutePath()
+
+  const transition = useTransition(location, {
+    key: pageRoutePath,
+    from: { opacity: 0, translateX: '-500px' },
+    enter: { opacity: 1, translateX: '0' },
+    leave: { opacity: 0, translateX: '500px' }
+  })
 
   return (
     <>
       <Navigation />
-      {matchRoutes(
-        [routes.home, <Home />],
-        [routes.articles, <Articles />],
-        [routes.articles.article, params => <Article {...{ params }} />],
-        [routes.notFound, params => <NotFound {...{ params }} />],
-      )}
+      {transition((props, transitionLocation) => {
+        const isLeaving = location !== transitionLocation
+        return (
+          <animated.div style={{ ...(isLeaving && { position: 'absolute' }), ...props }}>
+            {isLeaving
+              ? <StaticLocationProvider location={transitionLocation} children={<Content />} />
+              : <Content />
+            }
+          </animated.div>
```

Comment:
Ik zou de items met css grid over elkaar leggen, omdat dat van beide `Content` componenten de hoogte in acht wordt genomen, wat het makkelijker maakt om gekke scroll positieverspringen te fixen:

```css
.switch {
  display: grid;

  & > .route {
    min-width: 0;
    grid-row: 1 / -1;
    grid-column: 1 / -1;
  }
}
```
Hoe gaan we daar trouwens mee om? Naar boven scrollen on navigate? We zouden ook de positie bij moeten houden voor wanneer je de back button gebruikt.

In dat geval is de `isLeaving` denk ik ook niet meer nodig, omdat nieuwere items (volgensmij) door react spring lager in de DOM geplaatst worden en daardoor over de oude items heen liggen.

Reply:
@peeke Zou jij dat even in deze branch kunnen doen. Jij weet al hoe het moet en ik ga waarschijnlijk weer klooien.

Dat scrollen vind ik een prima ding om in een apart pull request op te pakken wanneer iemand daarmee aan de slag gaat. Het project waar dit voor nodig is heeft dat nog niet nodig.